### ISSUE TITLE
(Question ) management api - https 

### ISSUE THREAD

**Issue by San-eng** [guidesmiths/rascal#170]
Hi,
I want to configure the https agent to trust the private CA used to sign the server cert.
I think this configuration should be in the management.options section but I can't find the right option names. 
`                connection: {
                    heartbeat: 1,
                    protocol: "amqps",
                    hostname: "localhost",
                    user: "user",
                    password: "****",
                    port: 5671,
                    socketOptions: {
                        timeout: 1000,
                        ca: [capem]
                    },
                    management: {
                        slashes: true,
                        protocol: "https",
                        port: 15672,
                        options: {
                        },
                    },
                },`
Please can someone help me with this?

**Comment by San-eng** [comment]
Hi @cressie176 , thank you for your reply.
The application is not yet deployed to production but I'm considering using Rascal's management api feature for production environment to create the vhost if one doesn't exist. 
For this project ,  there is a requirement that all communications must be secure.

**Comment by cressie176** [comment]
Hi @San-eng, thanks for clarifying. 

Is it useful to create the vhost automatically without being able to create users & policies? I'd always assumed not, but might be missing something.

**Comment by cressie176** [comment]
I've added the ability to supply your own instance of superagent. You should be able to do something like the following.

```js
const superagent = require('superagent-defaults');
const ca = fs.readFileSync('ca.cert.pem');
const agent = superagent().ca(ca);

const components = { agent };
const broker = await Broker.create(config, components);
```

All of superagent's [TLS options](https://visionmedia.github.io/superagent/#tls-options) should be available.

Published in rascal@v14.1.0
